### PR TITLE
Field reset options

### PR DESCRIPTION
**Pull request by Bezmehrabi**
PR for #386.

### CODE REVIEW

File: src/FormController.js [hunk 1]
```diff
@@ -806,43 +806,65 @@ export class FormController {
...
+    const {
+      value,
+      resetError = true,
+      resetTouched = true,
+      resetDirt = true
+    } = options;
+
+    if (value) {
+      debug(`Resetting ${name}'s value to ${options.value}`);
+      ObjectMap.set(this.state.values, name, value);
```
Comment (joepuzzo) [inline src/FormController.js:824]: Would we not also want to call initializeValue here ? 🤔

File: __tests__/hooks/useFormApi.test.js [hunk 2]
```diff
@@ -284,4 +284,84 @@ describe('useFormApi', () => {
...
+    };
+
+
+    const { getByText } = render(
+      <Form formApiRef={formApiRef} initialValues={{ greeting: '123' }}>
+        <Input name="greeting" label="input1" validate={value => !!Number(value)} validateOn="change" />
+        <Button />
+      </Form>
+    );
+
```
Comment (joepuzzo) [inline __tests__/hooks/useFormApi.test.js:339]: Add assertion to check state before click please

File: __tests__/hooks/useFormApi.test.js [hunk 3]
```diff
@@ -284,4 +284,84 @@ describe('useFormApi', () => {
...
+    act(() => {
+      formApiRef.current.setValue('greeting', 'bar');
+    });
+
+    const button = getByText('Click Me');
+    fireEvent.click(button);
+
+    expect(formApiRef.current.getFormState()).toEqual(getState({
+      pristine: false,
+      dirty: true,
```
Comment (joepuzzo) [inline __tests__/hooks/useFormApi.test.js:349]: Should this not be pristine? Do I have bug in OG code maybe 🤔

File: __tests__/hooks/useFormApi.test.js [hunk 4]
```diff
@@ -284,4 +284,84 @@ describe('useFormApi', () => {
...
+      maskedValues: {
+        greeting: '123'
+      },
+      initialValues: {
+        greeting: '123',
+      },
+      valid: false,
+    }));
+
+  });
```
Comment (joepuzzo) [inline __tests__/hooks/useFormApi.test.js:366]: Can you add two more tests for the dirt and touched

File: __tests__/hooks/useFormApi.test.js [hunk 5]
```diff
@@ -284,4 +284,84 @@ describe('useFormApi', () => {
...
+      );
+    };
+
+    const { getByText } = render(
+      <Form formApiRef={formApiRef}>
+        <Input name="greeting" label="input1"/>
+        <Button />
+      </Form>
+    );
+
```
[the remark this answers is not in the document]
Comment (Bezmehrabi) [reply __tests__/hooks/useFormApi.test.js:305]: Done, 
I think the `pristine` should be `false` after resetting the value to `elon` 🤔.

File: __tests__/hooks/useFormApi.test.js (hunk 3 above)
Comment (Bezmehrabi) [reply __tests__/hooks/useFormApi.test.js:378]: we are changing the field value to `bar`.
based on the description in docs: ` A form is pristine when no values have changed`.
it's working correctly here I guess.

**Pull request merged**